packages/flutter_test/lib/src/binding.dart

```diff
@@ -488,19 +489,25 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
   /// events from the device).
   Offset localToGlobal(Offset point) => point;
 
+  // The source of the current pointer event.
+  //
+  // The [pointerEventSource] is set as the `source` parameter of
+  // [handlePointerEvent] and can be used in the immediate enclosing
+  // [dispatchEvent].
+  TestBindingEventSource _pointerEventSource = TestBindingEventSource.device;
```

Comment:
Looks like this property (and the logic for setting/retrieving it) can be moved into the `LiveTestWidgetsFlutterBinding` class since it is only needed there?

Reply:
Sadly not. `WidgetTester` calls the `handlePointerEvent` with source but `WidgetTester.binding` is of type `TestWidgetFlutterBinding`.